src/crimson/os/seastore/transaction.h

```diff
@@ -52,16 +52,23 @@ class Transaction {
 
   void add_to_retired_set(CachedExtentRef ref) {
     ceph_assert(!is_weak());
-    if (!ref->is_initial_pending()) {
+    if (ref->is_initial_pending()) {
+      ref->state = CachedExtent::extent_state_t::INVALID;
+      write_set.erase(*ref);
+      // XXX: if want to remove it from fresh_block_list to save
+      // writes, we will need to find a way to adjust relative paddrs
+    } else if (ref->is_mutation_pending()) {
+      ref->state = CachedExtent::extent_state_t::INVALID;
+      write_set.erase(*ref);
+      assert(ref->prior_instance);
+      retired_set.insert(ref->prior_instance);
+      assert(read_set.find(ref->prior_instance->get_paddr()) != read_set.end());
```

Comment:
nit, could use `set::count()` since we don't have `set::contains()` in C++17. like
```c++
assert(read_set.count(ref->prior_instance->get_paddr()));
```

more concise this way.